File: Atividade-pratica/atividade1.c
#include <stdio.h>
#include <string.h>

#define MAX_TAREFAS 100
#define TAM_NOME 100

typedef struct {
    char descricao[TAM_NOME];
    int concluida; // 0 = não, 1 = sim
} Tarefa;

void listarTarefas(Tarefa lista[], int total) {
    printf("\n--- Lista de Tarefas ---\n");
    for (int i = 0; i < total; i++) {
        printf("%d. [%c] %s\n", i + 1, lista[i].concluida ? 'X' : ' ', lista[i].descricao);
    }
    printf("-------------------------\n");
}

int main() {
    Tarefa tarefas[MAX_TAREFAS];
    int total = 0;
    int opcao;

    do {
        printf("\n1. Adicionar tarefa\n");
        printf("2. Concluir tarefa\n");
        printf("3. Listar tarefas\n");
        printf("0. Sair\n");
        printf("Escolha uma opção: ");
        scanf("%d", &opcao);
        getchar(); // limpar o buffer

        if (opcao == 1) {
            if (total < MAX_TAREFAS) {
                printf("Digite a descrição da tarefa: ");
                fgets(tarefas[total].descricao, TAM_NOME, stdin);
                tarefas[total].descricao[strcspn(tarefas[total].descricao, "\n")] = '\0'; // remover '\n'
                tarefas[total].concluida = 0;
                total++;
            } else {
                printf("Limite de tarefas atingido!\n");
            }
        } else if (opcao == 2) {
            int indice;
            listarTarefas(tarefas, total);
            printf("Digite o número da tarefa a concluir: ");
            scanf("%d", &indice);
            if (indice >= 1 && indice <= total) {
                tarefas[indice - 1].concluida = 1;
                printf("Tarefa marcada como concluída!\n");
            } else {
                printf("Número inválido.\n");
            }
        } else if (opcao == 3) {
            listarTarefas(tarefas, total);
        } else if (opcao == 0) {
            printf("Saindo...\n");
        } else {
            printf("Opção inválida!\n");
        }

    } while (opcao != 0);

    return 0;
}

File: Atividade-pratica/atividade1.py
# Aqui define o tamanho maximo de caracteres da minha tarefa e do tamanho do nome da tarefa
#define MAX_TAREFAS 100
#define TAM_NOME 100
max_tarefas = 100
tamanho_da_descricao = 100

# Crio minha variaveis
descricao = ''
total_de_tarefas = 0
tarefas = [] # Lista com dicionarios pois temos uma descrição e um verificador de concluida

def adicionar_tarefa(tarefa):
  item = {
    'descricao': tarefa,
    'concluida': 0
  }
  tarefas.append(item)


def listar_tarefas(lista):
  print('\n--- Lista de tarefas ---')
  if len(lista) > 0:
    for index, tarefa in enumerate(lista):
      print(f'{index + 1}. [{"X" if tarefa['concluida'] != 0 else ' '}] {tarefa['descricao']}')
  else:
    print('-------------------------\n')

while True:
  print('\n1. Adicionar tarefa')
  print('\n2. Concluir tarefa')
  print('\n3. Listar tarefas')
  print('\n0. Sair\n')
  opcao = input('Escolha uma opção: ')

  if opcao == '1':
    if total_de_tarefas < max_tarefas:
      descricao_tarefa = input('Digite a descricao da tarefa: ')
      descricao = descricao_tarefa[:tamanho_da_descricao]
      adicionar_tarefa(descricao)
      total_de_tarefas += 1
    else:
      print('Limite total de tarefas atingido!')

  elif opcao == '2':
    listar_tarefas(tarefas)
    concluir_tarefa = int(input('Digite o numero da tarefa a concluir: '))
  
    if concluir_tarefa >= 1 and concluir_tarefa <= total_de_tarefas:
      tarefas['concluida'] = 1 # PAREI AQUI, ESTA DANDO ERRO NESTA LINHA
      print('Tarefa marcada como concluída!\n')
    else:
      print('Numero inválido!')
  elif opcao == '3':
    listar_tarefas(tarefas)
  elif opcao == '0':
    print('Saindo... \n')
    break
  else:
    print('Opcao invalida!\n')